src/app/components/player/player.component.ts

```diff
@@ -0,0 +1,122 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, OnDestroy, OnInit } from '@angular/core';
+
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+import { Track } from '../../interfaces/track';
+import { PlayerService } from '../../services/player.service';
+
+@Component({
+  selector: 'app-player',
+  templateUrl: './player.component.html',
+  styleUrls: ['./player.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class PlayerComponent implements OnInit, OnDestroy {
+
+  track: Track;
+
+  // controls visualisation
+  isPlaying: boolean;
+  isFirst: boolean;
+  isLast: boolean;
+
+  audio: HTMLAudioElement;
+
+  private readonly destroyedSubject = new Subject<void>();
+
+  constructor(
+    private readonly playerService: PlayerService,
+    private readonly changeDetectorRef: ChangeDetectorRef
+  ) { }
+
+  ngOnDestroy(): void {
```

Comment:
Better to organize lifecycle hooks in proper order. For example:


```
ngOnInit
someInternalMethod
someSecondInternalMethod
ngOnDestroy
```

Reply:
It was on my linter rules 
`"member-ordering": [
      true,
      "public-before-private",
      "static-before-instance",
      "variables-before-functions"
    ]
`